handler/handler.go

```diff
@@ -75,6 +75,34 @@ func (h *Handler) Call(ev interface{}) {
 	}
 }
 
+// CallDirect is the same as Call, but only calls those event handlers that
+// listen for this specific event, i.e. that aren't interface handlers.
+func (h *Handler) CallDirect(ev interface{}) {
```

Comment:
Why do we need this?

Reply:
Without this, subscribers to `interface{}` events would both get e.g. a `GuildCreateEvent` and a `GuildJoinEvent`, effectively giving them the same event twice. This way only the `GuildJoinEvent` gets sent, as, in contrast to the `GuildCreateEvent` it gives situational information. `GuildCreateEvent`s can still be subscribed to, but directly, i.e. `AddHandler(func (*gateway.GuildCreateEvent))`.

Reply:
I think those handlers should be aware of this. This is mostly for event handlers that do its own event type switching, so we can't miss out a few.